pylint/checkers/base.py

```diff
@@ -2593,6 +2594,41 @@ def _check_type_x_is_y(self, node, left, operator, right):
         self.add_message("unidiomatic-typecheck", node=node)
 
 
+class ThreadingChecker(_BasicChecker):
+    """Checks for threading module
+
+    - useless with lock - locking used in wrong way that has no effect (with threading.Lock():)
+    """
+
+    msgs = {
+        "W9990": (  # TODO
+            "Useless usage of with statement and %s()",
+            "useless-with-lock",
+            "Used when a new lock instance is created by using with statement "
+            "which has no effect, instead of using existing instance to acquire lock. "
+            "Valid for various locking objects: Lock, RLock, Condition, Semaphore, BoundedSemaphore.",
```

Comment:
```suggestion
```
I think this might not need to be specified here and it also has the potential to become out of date if new lock are added later.
```suggestion
            "Valid for various locking objects: {', '.join(ThreadingChecker.LOCK_CLASSES)}.",
```

Reply:
Removed